File: app/src/main/java/com/softartdev/notecrypt/ui/main/MainPresenter.java
package com.softartdev.notecrypt.ui.main;

import com.softartdev.notecrypt.App;
import com.softartdev.notecrypt.db.DbStore;
import com.softartdev.notecrypt.model.Note;

import javax.inject.Inject;

import io.realm.RealmResults;

public class MainPresenter {
    private MainView mView;

    @Inject
    DbStore dbStore;

    MainPresenter(MainView view) {
        App.getAppComponent().inject(this);
        mView = view;
    }

    void updateNotes() {
        RealmResults<Note> notes = dbStore.getNotes();
        mView.onUpdateNotes(notes);
    }

    void addNote() {
        mView.onAddNote();
    }
}

File: app/src/main/java/com/softartdev/notecrypt/ui/main/MainPresenter.kt
package com.softartdev.notecrypt.ui.main

import com.softartdev.notecrypt.data.DataManager
import com.softartdev.notecrypt.di.ConfigPersistent
import com.softartdev.notecrypt.ui.base.BasePresenter
import javax.inject.Inject

@ConfigPersistent
class MainPresenter @Inject
constructor(private val dataManager: DataManager) : BasePresenter<MainView>() {

    fun updateNotes() {
        checkViewAttached()
        addDisposable(dataManager.notes()
                .subscribe(
                        {notes -> mvpView?.onUpdateNotes(notes) }
                        , { it.printStackTrace() }))
    }
}
